File: 2340-minimum-adjacent-swaps-to-make-a-valid-array/2340-minimum-adjacent-swaps-to-make-a-valid-array.java
class Solution {
    public int minimumSwaps(int[] nums) {
        if(nums.length <2){
            return 0;
        }
        int maxVal = nums[0];
        int minVal = nums[0];
        int maxIndex = 0;
        int minIndex = 0;
        for(int i=1;i<nums.length;i++){
            if(maxVal<=nums[i]){
                maxIndex = i;
                maxVal = nums[i];
            }
        }
        
        for(int i=1;i<nums.length;i++){
            if(minVal>nums[i]){
                minIndex = i;
                minVal = nums[i];
            }
        }
        // cout<<maxIndex<<" "<<minIndex<<endl;
        System.out.println(maxIndex+" "+minIndex);
        
        
        int ans = nums.length - maxIndex-1;
        
        ans = ans + minIndex;
        if(maxIndex<minIndex){
            ans--;
        }
        
        return ans;
    }
}

File: 2340-minimum-adjacent-swaps-to-make-a-valid-array/2340-minimum-adjacent-swaps-to-make-a-valid-array.cpp
class Solution {
public:
    int minimumSwaps(vector<int>& nums) {
        if(nums.size() <2){
            return 0;
        }
        int n = nums.size();
        int maxVal= *max_element(nums.begin(),nums.end());
        int minVal = *min_element(nums.begin(),nums.end());
        int minIndex  =0;
        int maxIndex = nums.size()-1;
        for(int i=0;i<nums.size();i++){
            if(nums[i] == minVal){
                minIndex = i;
                break;
            }
        }
        
        for(int i=nums.size()-1;i>=0;i--){
            if(nums[i] == maxVal){
                maxIndex = i;
                break;
            }
        }
        int ans = minIndex+ n - maxIndex-1;
        
        if(maxIndex<minIndex){
            ans--;
        }
    
        return ans;
    }
    
};